apps/choleskey/cholesky_tiled.cpp

```diff
@@ -0,0 +1,149 @@
+
+2. Therefore, the first step is to implement tiled Cholesky decomposition algorithm.
+
+3. This file is to implement tiled Cholesky decomposition algorithm. 
+    reference the implementation from Intel open source project, hetero-streams which will no longer be maintained by Intel.
+    https://github.com/intel/hetero-streams/tree/master/ref_code/cholesky
+
+4. Additionally, include openblas library when build:
+   or $ export OPENBLAS_DIR=/openblas/path
+*/
+
+#include <bits/stdc++.h>
+#include <cblas.h>
+#include <lapacke.h>
+#include <omp.h>
+#include <algorithm>
+#include <cmath>
+#include <cstdlib>
+#include <cstring>
+#include "tiled_cholesky_help.hpp"
+
+using data_type = double;
+
+void tiled_cholesky(double* matrix_split[], const int tile_size, const int num_tiles, CBLAS_ORDER blasLay,
+                    const int lapackLay) {
+    unsigned int m, n, k;
+
+    for (k = 0; k < num_tiles; ++k) {
+        //POTRF
+        int info = LAPACKE_dpotrf(LAPACK_ROW_MAJOR, 'L', tile_size, matrix_split[k * num_tiles + k], tile_size);
+
+        for (m = k + 1; m < num_tiles; ++m) {
+            //DTRSM
+            cblas_dtrsm(blasLay, CblasRight, CblasLower, CblasTrans, CblasNonUnit, tile_size, tile_size, 1.0,
+                        matrix_split[k * num_tiles + k], tile_size, matrix_split[m * num_tiles + k], tile_size);
+        }
+
+        for (n = k + 1; n < num_tiles; ++n) {
+            //DSYRK
+            cblas_dsyrk(blasLay, CblasLower, CblasNoTrans, tile_size, tile_size, -1.0, matrix_split[n * num_tiles + k],
+                        tile_size, 1.0, matrix_split[n * num_tiles + n], tile_size);
+
+            for (m = n + 1; m < num_tiles; ++m) {
+                //DGEMM
+                cblas_dgemm(blasLay, CblasNoTrans, CblasTrans, tile_size, tile_size, tile_size, -1.0,
+                            matrix_split[m * num_tiles + k], tile_size, matrix_split[n * num_tiles + k], tile_size, 1.0,
+                            matrix_split[m * num_tiles + n], tile_size);
+            }
+        }
+    }
+}
+
+int main(int argc, char** argv) {
+
+    // TODO : introduce args
+    int matrix_size, num_tiles, tile_size, tot_tiles;
+    matrix_size = 40;  // must be an input
+    num_tiles = 2;     // matrix size MUST be divisible
+    int verifycorrectness = 1;
+    bool layRow = true;
+
+    fmt::print("matrix_size = {}, num_tiles = {}\n\n", matrix_size, num_tiles);
+
+    // Check mat_size is divisible by num_tiles
+    if (matrix_size % num_tiles != 0) {
+        fmt::print("matrix size must be divisible by num_tiles.. aborting\n");
+        throw std::invalid_argument("Matrix size is not divisible by num_tiles");
+    }
+
+    if (matrix_size == 0) {
+        fmt::print(" 0 illegal input matrix_size \n");
+        std::exit(0);
+    }
+
+    tile_size = matrix_size / num_tiles;
+    tot_tiles = num_tiles * num_tiles;
+
+    double* A = new double[matrix_size * matrix_size];
```

Comment:
Please use `data_type` instead of `double` where applicable to ensure portability. Thank you.